modules/core/arrow-typeclasses/src/main/kotlin/arrow/typeclasses/MonadContinuations.kt

```diff
@@ -12,8 +13,8 @@ interface BindingInContextContinuation<in T> : Continuation<T> {
 }
 
 @RestrictsSuspension
-open class MonadContinuation<F, A>(M: Monad<F>, override val context: CoroutineContext = EmptyCoroutineContext) :
-  Continuation<Kind<F, A>>, Monad<F> by M {
+open class MonadContinuation<F, A>(val M: Monad<F>, override val context: CoroutineContext = EmptyCoroutineContext) :
```

Comment:
I'd say extract a couple of interfaces from MonadContinuation and MCError, and have this as the default implementation. That way we'll make it a typeclass instead of a class where users have full access to implementation details.

Reply:
Tried that in the original implementation but the issue is that `A` needs to be carried around always because its the `A` from `Continuation<A>` injected by the compiler in suspended functions so all attempts to turn it into a type class are frustrating since there will always be `F` and `A` in the top signature. If anyone else wants to give it another shot I'll be happy to help.